Define column names (#6140)
dbt-labs/docs.getdbt.com: website/docs/docs/build/snapshots.md

</details>

<VersionBlock firstVersion="1.9">

> ### Define column names in snapshot table

Reviewer: I don't think this is the correct language.. something more like "customize the names of the dbt-generated meta fields in your snapshot tables"

> ### Define column names in snapshot table

Author: ```suggestion
### Customize names of dbt-generated meta fields in snapshot tables 
```

> 

Reviewer 2: small sugg that's not a blocker: maybe have the header outside the versionblock and tell users on 1.8 and lower that this feature is only avail in 1.9 and higher or versionless so they know?

Reviewer: ++ we need to do this for all new features!

> In dbt Cloud, there's a config which will now allow you to define the names of the columns that are created in the [snapshot table](/docs/build/snapshots#snapshot-meta-fields). Users can now set this config to use different names.

Reviewer 2: i'm a little confused -- is this only for dbt cloud? i thought this was something avail to both cloud and core users since it relates to snapshots? maybe the diff is it's available to versionless users now and 1.9 users when 1.9 is launched?

Author: Sorry - that was a force if habit. I'll update this. Thanks for this

> In dbt Cloud, there's a config which will now allow you to define the names of the columns that are created in the [snapshot table](/docs/build/snapshots#snapshot-meta-fields). Users can now set this config to use different names.

Reviewer 2: ```suggestion
Define the names of the columns created in the [snapshot table](/docs/build/snapshots#snapshot-meta-fields) using custom configurations. The original columns were named `dbt_valid_to`, `dbt_valid_from`, `dbt_scd_id`, and `dbt_updated_at`, however, users can now customize these names as needed.
```

> In dbt Cloud, there's a config which will now allow you to define the names of the columns that are created in the [snapshot table](/docs/build/snapshots#snapshot-meta-fields). Users can now set this config to use different names.

Author: ```suggestion
In [Versionless](/docs/dbt-versions/upgrade-dbt-version-in-cloud#versionless), you can now define the names of the columns created in the [snapshot table](/docs/build/snapshots#snapshot-meta-fields) using custom configurations. The original columns were named `dbt_valid_to`, `dbt_valid_from`, `dbt_scd_id`, and `dbt_updated_at`, however, users can now customize these names as needed.
```

> In [Versionless](/docs/dbt-versions/upgrade-dbt-version-in-cloud#versionless), you can now define the names of the columns created in the [snapshot table](/docs/build/snapshots#snapshot-meta-fields) using custom configurations. The original columns were named `dbt_valid_to`, `dbt_valid_from`, `dbt_scd_id`, and `dbt_updated_at`, however, users can now customize these names as needed.

Reviewer: ```suggestion
In [Versionless](/docs/dbt-versions/upgrade-dbt-version-in-cloud#versionless), you can now customize the names of the meta columns created by dbt in the [snapshot table](/docs/build/snapshots#snapshot-meta-fields) using the `snapshot_meta_column_names` configuration. 
```

> 

Reviewer: Let's add an example

> If a user renames the columns for a snapshot without updating an existing table, they will get an error. The recommendation is to use these settings for new snapshots, or to arrange to update your existing tables prior to committing a column name change.

Reviewer 2: ```suggestion
Note, if you rename snapshot columns without updating the existing table, it will cause an error. It's recommended to use this feature for new snapshots or update your existing tables before committing a column name change.
```

> 

Author: ```suggestion
This feature will also be available in the upcoming 1.9 release. 
```

> This feature will also be available in the upcoming 1.9 release. 

Reviewer: ```suggestion
This feature will also be available in the upcoming 1.9 release of dbt Core. 
```
